test/test_e2e.rb

```diff
@@ -104,4 +104,10 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert queue.fail_fast_limit_reached?
```

Comment:
Also assert that the processed jobs are the expected ones (and no more!) and that `#build_successful?` returns false. There are many things that could go wrong there.